Fix to Polish town names generation

katafrakt opened the pull request:
## Motivation / Problem

Polish name generation has a small bug which results in grammatically incorrect names.

<!--
Describe here shortly
* For bug fixes:
    * What problem does this solve?
    * If there is already an issue, link the issue, otherwise describe the problem here.
* For features or gameplay changes:
    * What was the motivation to develop this feature?
    * Does this address any problem with the gameplay or interface?
    * Which group of players do you think would enjoy this feature?
-->


## Description

The name "Susz" is masculine, not neuter. The correct associations are "Susz Morski", "Susz Górny", not "Susz Morskie" or "Susz Górne" like now in the game.

<!--
Describe here shortly
* For bug fixes:
    * How is the problem solved?
* For features or gameplay changes:
    * What does this feature do?
    * How does it improve/solve the situation described under 'motivation'.
-->


## Limitations

<!--
Describe here
* Is the problem solved in all scenarios?
* Is this feature complete? Are there things that could be added in the future?
* Are there things that are intentionally left out?
* Do you know of a bug or corner case that does not work?
-->


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

katafrakt committed:
Fix: Replace Susz with Leszno

Susz is masculine, not neuter, so it should result in "Susz Mazowiecki",
"Susz Morski", and not "Susz Mazowieckie" or "Susz Morskie". However,
because order of the names whould not be changed, it was replaced with
Leszno, which is neuter.

PeterN commented:
Problem with moving parts like this is it changes the names of towns within existing games. If it must be changed, it would be better to replace this part with something else that can stay in the same place.

katafrakt commented:
I see. I'll figure out some replacement.

Does that mean that town names are basically unchangeable? I'm asking because Polish neuter names are generally overrepresented, compared to reality.

glx22 commented:
Yes they are unchangeable. Maybe one day it will be different, but not yet.

katafrakt committed:
Fix: Correct name of Golub-Dobrzyn

katafrakt commented:
Okay, I replaced Susz with [Leszno](https://en.wikipedia.org/wiki/Leszno) (which is neuter) and also fixed a name of [Golub-Dobrzyń](https://en.wikipedia.org/wiki/Golub-Dobrzy%C5%84).

2TallTyler commented:
For reference, here's a discussion about one possible way to fix town names: https://github.com/OpenTTD/OpenTTD/discussions/8737
